src/hcf-release/src/acceptance-tests-brain/test-scripts/008_cf_usb_test.sh

```diff
@@ -0,0 +1,199 @@
+#!/bin/bash
+# # ## ### ##### ######## #############  #####################
+## Explanations and overview:
+
+## This test does a lot of setup to determine that the universal
+## service broker (cf-usb) is actually working.
+#
+## 0. A domain for TCP routing is created, to connect all the pieces.
+#
+## 1. A local mysql server is started as an app and made available
+##    through cf TCP routing.
+#
+## 2. The mysql sidecar is started as an app, configured to talk to
+##    the mysql server from (1).
+#
+## 3. The cf-usb is configured to talk to and use the side car.
+#
+## 4. Then we can check that mysql appears in the marketplace, create
+##    a service from it, and check that this service is viewable too.
+#
+## In the code below these phases are marked with "--(N)--" where N is
+## the step number.
+#
+## Note, the applications of step 1 and 2 are docker apps. This is why
+## the `pre-start.sh` script enables the `diego_docker` feature-flag
+## of CF. For step (3) the `pre-start.sh` script extended the `cf`
+## client with the `cf-usb-plugin` plugin.
+
+function get_port
+{
+    port=$(awk '/Route .* has been created/ {print $2}' < ${TMP}/$1 | cut -f 2 -d ':')
+    if [ -z "${port}" ]; then
+	echo "ERROR: Could not determine the assigned random port number for $1"
+	echo "ERROR: Mapping route to random port failed for $1"
+	exit 1
+    fi
+    echo $port
+}
+
+function wait_on_port
+{
+    for (( i = 0; i < 12 ; i++ )) ; do
+	if curl --fail -s -o /dev/null ${CF_TCP_DOMAIN}:$1 ; then
+            break
+	fi
+	sleep 5
+    done
+    # Last try, any error will abort the test
+    curl ${CF_TCP_DOMAIN}:$1
+}
+
+## # # ## ### Tracing and common configuration ### ## # #
+
+set -o errexit
+set -o xtrace
+
+function random_suffix { head -c2 /dev/urandom | hexdump -e '"%04x"'; }
+CF_ORG=${CF_ORG:-org}-$(random_suffix)
+CF_SPACE=${CF_SPACE:-space}-$(random_suffix)
+CF_TCP_DOMAIN=${CF_TCP_DOMAIN:-tcp-$(random_suffix).${CF_DOMAIN}}
+
+## # # ## ### Login & standard entity setup/cleanup ### ## # #
+
+function login_cleanup() {
+    trap "" EXIT ERR
+    set +o errexit
+
+    cf delete-space -f ${CF_SPACE}
+    cf delete-org -f ${CF_ORG}
+
+    set -o errexit
+}
+trap login_cleanup EXIT ERR
+
+# target, login, create work org and space
+cf api --skip-ssl-validation api.${CF_DOMAIN}
+cf auth ${CF_USERNAME} ${CF_PASSWORD}
+
+cf create-org ${CF_ORG}
+cf target -o ${CF_ORG}
+
+cf create-space ${CF_SPACE}
+cf target -s ${CF_SPACE}
+
+## # # ## ### Test-specific configuration ### ## # #
+
+# Location of the test script. All other assets will be found relative
+# to this.
+SELFDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
```

Comment:
Do we need to deal with `CDPATH` fun here?

Reply:
AFAIK no. Also standard code inherited IIRC, or at least the same for all testcases using a SELFDIR.
If we want to change, IMHO a separate ticket.

Reply:
Confirmed identical across all test files.